website/frontend/src/context/alert/Provider.tsx

```diff
@@ -1,26 +1,150 @@
+import { DEFAULT_INFO_OPTIONS, DEFAULT_ERROR_OPTIONS, DEFAULT_LOADING_OPTIONS, DEFAULT_SUCCESS_OPTIONS } from './const';
+import { AlertTimer, AlertInstance, AlertOptions, TemplateAlertOptions, AlertContainerFactory } from './types';
+import { AlertContext } from './Context';
+
 import { AlertTemplate } from 'components/AlertTemplate';
+import { Wrapper, Transition } from 'components/Alert';
 
-type Props = {
-  children: ReactNode;
-};
+//Consider integrating react-toastify
+const AlertProvider = ({ children }: Props) => {
+  const root = useRef<HTMLDivElement | null>(null);
 
-const options: AlertOptions = {
-  position: positions.BOTTOM_RIGHT,
-  timeout: 10000,
-  transition: transitions.FADE,
-  containerStyle: {
-    zIndex: ZIndexes.alert,
-    width: '390px',
-    left: 'unset',
-    right: '22px',
-    bottom: '5px',
-    alignItems: 'stretch',
-  },
-};
+  const timers = useRef<AlertTimer>(new Map());
+  const [alerts, setAlerts] = useState<AlertInstance[]>([]);
+
+  const createTimer = useCallback((alertId: string, callback: (alertId: string) => void, timeout) => {
+    if (timeout > 0) {
+      const timerId = setTimeout(() => callback(alertId), timeout);
+
+      timers.current.set(alertId, timerId);
+    }
+  }, []);
+
+  const removeTimer = useCallback((alertId: string) => {
+    const timer = timers.current.get(alertId);
+
+    if (timer) {
+      clearTimeout(timer);
+      timers.current.delete(alertId);
+    }
+  }, []);
+
+  const remove = useCallback(
+    (id: string) => {
+      removeTimer(id);
+      setAlerts((prevState) => prevState.filter((alert) => alert.id !== id));
+    },
+    [removeTimer]
+  );
+
+  const show = useCallback(
+    (content: ReactNode, options: AlertOptions): string => {
+      const id = nanoid(8);
+
+      createTimer(id, remove, options.timeout);
+      setAlerts((prevState) => [
+        ...prevState,
+        {
+          id,
+          content,
+          options,
+        },
+      ]);
+
+      return id;
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [createTimer, remove]
+  );
 
-const AlertProvider = ({ children }: Props) => <Provider template={AlertTemplate} children={children} {...options} />;
+  const update = useCallback(
+    (id: string, content: ReactNode, options?: Omit<AlertOptions, 'castomId'>) => {
+      let updatedAlert: AlertInstance;
+
+      removeTimer(id);
+
+      setAlerts((prevState) =>
+        prevState.map((alert) => {
+          if (alert.id !== id) {
+            return alert;
+          }
+
+          return (updatedAlert = {
+            id,
+            content,
+            options: {
+              ...alert.options,
+              ...options,
+            },
+          });
+        })
+      );
+
+      if (updatedAlert!) {
+        createTimer(updatedAlert.id, remove, updatedAlert.options.timeout);
+      }
+    },
+    [removeTimer, createTimer, remove]
+  );
+
+  const сreateTemplateAlert = useCallback(
```

Comment:
```suggestion
  const getAlertTemplate = useCallback(
```
<img width="770" alt="image" src="https://user-images.githubusercontent.com/67055679/169914160-88926d01-d277-4b2c-898c-18beab86b518.png">
The first 'c' letter right now is written in cyrillic